### PR TITLE
unit tests developed and some code refactored #887 

### PR DESCRIPTION
**Pull request by RomanKlymus**
Just added some unit tests and refactor code

### CODE REVIEW

File: src/main/java/com/softserve/teachua/dto/category/SuccessCreatedCategory.java [hunk 1]
```diff
@@ -12,6 +12,8 @@
 @Data
 public class SuccessCreatedCategory implements Convertible {
 
+    private Long id;
+
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/dto/category/SuccessCreatedCategory.java:16]: add this to show the user which id in the created category

File: src/main/java/com/softserve/teachua/repository/CategoryRepository.java [hunk 2]
```diff
@@ -21,12 +21,12 @@
...
+    List<Category> findInSortedOrder();
+
     Page<Category> findAll(Pageable pageable);
 
     @Query(value = "SELECT *  FROM categories AS c WHERE LOWER(c.name) LIKE LOWER('%' || :text || '%') ORDER BY RANDOM() LIMIT 3",
             nativeQuery = true)
     List<Category> findRandomTop3ByName(@Param("text") String enteredText);
-
-    @Query(value = "select * FROM categories AS c ORDER BY sortBy DESC LIMIT 1 ", nativeQuery = true)
-    Category findLastCategory();
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/repository/CategoryRepository.java:31]: removed unused query

File: src/main/java/com/softserve/teachua/service/impl/CategoryServiceImpl.java [hunk 3]
```diff
@@ -111,17 +111,6 @@ public SuccessCreatedCategory addCategory(CategoryProfile categoryProfile) {
...
-        if(categoryProfile.getSortby() == null){
-            Category lastCategory = categoryRepository.findLastCategory();
-            Integer currentLastSortNumber = lastCategory.getSortby();
-
-            categoryProfile.setSortby(currentLastSortNumber);
-
-            lastCategory.setSortby(currentLastSortNumber+=10);
-            categoryRepository.save(lastCategory);
-        }
-
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/service/impl/CategoryServiceImpl.java:124]: removed useless code

File: src/main/java/com/softserve/teachua/service/impl/CategoryServiceImpl.java [hunk 4]
```diff
@@ -134,11 +123,14 @@ public SuccessCreatedCategory addCategory(CategoryProfile categoryProfile) {
...
     public List<CategoryResponse> getAllCategories() {
-        List<CategoryResponse> categoryResponses = categoryRepository.findAll()
-                .stream()
-                .map(category -> (CategoryResponse) dtoConverter.convertToDto(category, CategoryResponse.class))
-                .sorted(Comparator.comparing(CategoryResponse::getSortby))
-                .collect(Collectors.toList());
+        List<Category> categoryList = categoryRepository.findInSortedOrder();
+        if (categoryList == null) {
+            return Collections.emptyList();
+        }
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/service/impl/CategoryServiceImpl.java:129]: check does repository return null, if it is return empty list

File: src/main/java/com/softserve/teachua/service/impl/CategoryServiceImpl.java [hunk 5]
```diff
@@ -134,11 +123,14 @@ public SuccessCreatedCategory addCategory(CategoryProfile categoryProfile) {
      */
     @Override
     public List<CategoryResponse> getAllCategories() {
-        List<CategoryResponse> categoryResponses = categoryRepository.findAll()
-                .stream()
-                .map(category -> (CategoryResponse) dtoConverter.convertToDto(category, CategoryResponse.class))
-                .sorted(Comparator.comparing(CategoryResponse::getSortby))
-                .collect(Collectors.toList());
+        List<Category> categoryList = categoryRepository.findInSortedOrder();
```
Comment (RomanKlymus) [inline src/main/java/com/softserve/teachua/service/impl/CategoryServiceImpl.java:126]: move sort functional to database query

**Review by VictoriaAmazonka — APPROVED** [review]

**Pull request merged**